src/superphot_plus/ztf_transient_fit.py

```diff
@@ -319,67 +308,79 @@ def flux_model_smooth(t_data, A, beta, gamma, t0, tau_rise, tau_fall):
 
         return f_model
 
-    popt_r, pcov = curve_fit(  # pylint: disable=unused-variable
-        flux_model_smooth,
-        tdata[bdata == "r"],
-        fdata[bdata == "r"],
-        p0=p0[:6],
-        sigma=ferrdata[bdata == "r"],
-        bounds=bounds,
-        maxfev=100000,
-    )
+    opts = {}
 
-    p0_g = popt_r * p0[6:]
-    p0_g[2] = popt_r[2] + p0[8]
-    p0_g[5] = popt_r[5] + p0[-1]
-    p0_g[4] = popt_r[4] + p0[-2]
+    ## Get parameters for the reference band.
+    priors = mb_priors.bands[ref_band]
 
-    popt_g, pcov = curve_fit(
+    prior_array = [
+        max_flux,
+        priors.beta.mean,
+        priors.gamma.mean,
+        max_flux_loc,
+        priors.tau_rise.mean,
+        priors.tau_fall.mean,
+    ]
+
+    ref_results, _ = curve_fit(  # pylint: disable=unused-variable,unbalanced-tuple-unpacking
         flux_model_smooth,
-        tdata[bdata == "g"],
-        fdata[bdata == "g"],
-        p0=p0_g,
-        sigma=ferrdata[bdata == "g"],
+        tdata[bdata == ref_band],
+        fdata[bdata == ref_band],
+        p0=prior_array,
+        sigma=ferrdata[bdata == ref_band],
         bounds=bounds,
         maxfev=100000,
     )
 
-    print(popt_g, popt_r)
+    opts[ref_band] = ref_results
 
-    prefix = fn.split("/")[-1][:-4]
+    for band, priors in mb_priors.bands.items():
+        if band == ref_band:
+            continue
 
-    plt.errorbar(
-        tdata[bdata == "g"],
-        fdata[bdata == "g"],
-        yerr=ferrdata[bdata == "g"],
-        c="g",
-        label="g",
-        fmt="o",
-    )
-    plt.errorbar(
-        tdata[bdata == "r"],
-        fdata[bdata == "r"],
-        yerr=ferrdata[bdata == "r"],
-        c="r",
-        label="r",
-        fmt="o",
-    )
+        prior_array = [
+            priors.amp.mean,
+            priors.beta.mean,
+            priors.gamma.mean,
+            priors.t_0.mean,
+            priors.tau_rise.mean,
+            priors.tau_fall.mean,
+        ]
+        prior_array = prior_array * ref_results
+
+        popt, _ = curve_fit(  # pylint: disable=unused-variable,unbalanced-tuple-unpacking
+            flux_model_smooth,
+            tdata[bdata == band],
+            fdata[bdata == band],
+            p0=prior_array,
+            sigma=ferrdata[bdata == band],
+            bounds=bounds,
+            maxfev=100000,
+        )
+
+        opts[band] = popt
```

Comment:
maybe revert the aux band params to popt / ref_popt before outputs so they can more easily be compared with the other sampling techniques